src/test/java/com/epam/page/object/generator/integrationalTests/MainIntegrationTest.java

```diff
@@ -0,0 +1,401 @@
+        };
+
+        if (manualFields.size() != 0 && testFields.size() != 0) {
+
+            manualFields.sort(fieldsComparator);
+            testFields.sort(fieldsComparator);
+
+            Iterator<Field> testIt = testFields.iterator();
+            Iterator<Field> manualIt = manualFields.iterator();
+
+            while (testIt.hasNext()) {
+                Field testCurrentField = testIt.next();
+                Field manualCurrentField = manualIt.next();
+
+                List<Annotation> testFieldAnnotations = Arrays
+                    .asList(testCurrentField.getDeclaredAnnotations());
+                List<Annotation> manualFieldAnnotations = Arrays
+                    .asList(manualCurrentField.getDeclaredAnnotations());
+                assertEquals("Different field annotations", testFieldAnnotations,
+                    manualFieldAnnotations);
+
+                int testFieldModifiers = testCurrentField.getModifiers();
+                int manualFieldModifiers = manualCurrentField.getModifiers();
+                assertEquals("Different field modifiers", testFieldModifiers,
+                    manualFieldModifiers);
+
+                String[] newTestName = testCurrentField.toString().split(" ");
+                String[] newManualName = manualCurrentField.toString().split(" ");
+                String testImportForField = newTestName[1];
+                String manualImportForField = newManualName[1];
+                assertEquals("Different import for field", testImportForField,
+                    manualImportForField);
+                assertEquals("Different field names", testCurrentField.getName(),
+                    manualCurrentField.getName());
+            }
+        }
+    }
+
+    /**
+     * Gets case dirs names. For example: in src/test/manual dir, test case path will be:
+     * src/test/manual/TEST_CASE_DIR. Every case dir name is used for searching
+     * manual classes in this dir
+     *
+     * @return list of test cases dirs name
+     */
+    private static List<Path> getCaseDirsNames(String searchingDir, int depth) throws IOException {
+        Pattern pattern = Pattern.compile("manual([\\\\/])[a-zA-Z]+$");
+        return
+            Files.find(
+                Paths.get(searchingDir),
+                depth,
+                (path, basicFileAttributes) ->
+                    pattern.matcher(path.toString()).find()
+            ).collect(Collectors.toList());
+    }
+
+    /**
+     * Sorts list of TestClassesDTO objects. It's necessary, because all classes of test case
+     * can have dependencies of classes which represents pages. So all classes from page subdir
+     * have to be tested first.
+     *
+     * @param classesList list of test case classes to sort
+     * @return sorted list
+     */
+    private static List<TestClassesDTO> sortClassesList(List<TestClassesDTO> classesList) {
+        classesList.sort(
```

Comment:
Comparator makes code harder to read, following snippet some easier:
    
        Map<Boolean, List<TestClassesDTO>> grouped = classesList.stream()
                .collect(Collectors.groupingBy(t -> t.getManualClassFilePath().contains("page")));
        ArrayList<TestClassesDTO> result = new ArrayList<>(grouped.get(true));
        result.addAll(grouped.get(false));
        return result;